wrapanapi/systems/redfish.py

```diff
@@ -4,27 +4,124 @@
 """
 from __future__ import absolute_import
 
+import redfish_client
+
+from wrapanapi.entities.server import Server, ServerState
 from wrapanapi.systems.base import System
 
 
+class RedfishServer(Server):
+    state_map = {
+        'On': ServerState.ON,
+        'Off': ServerState.OFF,
+        'PoweringOn': ServerState.POWERING_ON,
+        'PoweringOff': ServerState.POWERING_OFF,
+    }
+
+    def __init__(self, system, raw=None, **kwargs):
+        """
+        Constructor for RedfishServer.
+
+        Args:
+            system: RedfishSystem instance
+            raw: the root resource in the Redfish API
+            odata_id: (optional) the @odata.id reference of this instance
+        """
+        self._odata_id = raw['@odata.id'] if raw else kwargs.get('odata_id')
+        if not self._odata_id:
+            raise ValueError("missing required kwargs: 'odata_id'")
+
+        super(RedfishServer, self).__init__(system, raw, **kwargs)
+
+    @property
+    def server_cores(self):
+        """Return the number of cores on this server."""
+        return sum([int(p.TotalCores) for p in self.raw.Processors.Members])
+
+    @property
+    def server_memory(self):
+        """Return the amount of memory on the server, in MiB."""
+        return self.raw.MemorySummary.TotalSystemMemoryGiB * 1024
+
+    @property
+    def state(self):
+        """Retrieve the current power status of the physical server."""
+        return self.raw.PowerState
+
+    def _get_state(self):
+        """
+        Return ServerState object representing the server's current state.
+
+        The caller should call self.refresh() first to get the latest status
+        from the API.
+        """
+        return self._api_state_to_serverstate(self.state)
+
+    def _identifying_attrs(self):
```

Comment:
IMAO it would be better if we didn't copy-paste docstring from the interface, because we only pollute the code. I'd only add comments if we do something special.

Reply:
Please remove redundant docstrings in all other interface implemetations as well.

Reply:
Actually those were for your benefit. :D
👍  I'll remove them.

Reply:
PyCharm shows me a "go to ancestor definition" for each method so the documentation is always just a click away ;)

Reply:
Yes, but not everybody uses PyCharm. I put the comments back because they help with code understanding more than bother with too much text.